pkg/config/config.go

```diff
@@ -145,3 +150,49 @@ func configFileWriter() (io.WriteCloser, error) {
 	}
 	return f, nil
 }
+
+const depthQueryParam = int32(5)
+
+type Client struct {
+	CloudClient       *cloudv6.APIClient
+	AuthClient        *sdkgoauth.APIClient
+	CertManagerClient *certmanager.APIClient
+	DbaasClient       *dbaas.APIClient
+}
+
+func NewClient(name, pwd, token, hostUrl string) (*Client, error) {
```

Comment:
Ideally, this function shouldn't be visible outside of this package - it should be illegal to be able to bypass the `GetClient` func (which is the one making sure there is never more than one instance of a Client). That being said though, I am aware the tests are very resistant to change, so if you think un-exporting `NewClient` is too much effort (because of the implied effort in changing the tests to use the `GetClient` func), we can do this part in some later PR when the tests become easier to manage

Reply:
As a compromise, I think we can add a comment to prevent future devs from ever using this function in new code, and to point them toward `GetClient` instead